packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -215,7 +216,13 @@ class DoubleTapGestureRecognizer extends GestureRecognizer {
           return false;
       }
     }
-    return super.isPointerAllowed(event);
+
+    // If second tap is not allowed, reset the state.
+    final bool isPointerAllowed = super.isPointerAllowed(event);
```

Comment:
With this change, should we also remove the switch case above that checks for `kPrimaryButton` but keep the check for callbacks? That would let `DoubleTapGestureRecognizer` work for any `button` and `allowedButtonsFilter` can be used to filter that. We might want to set the default for `DoubleTapGestureRecognizer.allowedButtonsFilter` to be `kPrimaryButton` as that the recognizer only previously accepted `kPrimaryButton` as its default behavior.

Reply:
Sure, I'm going to try. I hate the tests 😭.